assets/src/edit-story/app/story/storyProvider.js

```diff
@@ -35,6 +36,7 @@ function StoryProvider( { storyId, children } ) {
 	// Story state is stored in these three immutable variables only!
 	// Don't update 1 of these in an effect based off another base variable.
 	// Only update these directly as a response to user or api interactions.
+	const [ designMode, setDesignMode ] = useState( DesignMode.DESIGN );
```

Comment:
Wondering if the `designMode` should be part of the `storyProvider` as it is now or if it would make sense in the canvas level? Is the design state relevant for a Page only?

Reply:
It's needed in the sidebar as well of course, just realized.

Reply:
@miina I think your intuition is more correct here and maybe we should have the canvas state to sit above the sidebar? I don't think they are ever out of sync. Or maybe it's a higher-level state altogether - editor-wide state.

Reply:
If we could use it for the grid view as well (#3836) then it seems more like an editor-wide state.

Reply:
As we do need some kind of editor-wide flag to indicate _grid-view-mode_, this seems like a great candidate for it.

However I suggest that we implement it with a strategy pattern and create different "editor view strategies". Then all relevant components would merely check the active editor view strategy. Thus e.g. the inspector panel doesn't need to know whether or not a certain design mode includes the inspector or not - it'll just refer to the strategy and update accordingly.

In implementation, it would be something like:

```js
const EDITOR_GRID = {
  hasInspector: false,
  hasLibrary: false,
  hasMoveable: false,
  hasPage: false,
  ...
};
const EDITOR_DESIGN = {
  hasInspector: true,
  hasLibrary: true,
  hasMoveable: true,
  hasPage: true,
  ...
};
const DESIGN_REPLAY = {
  hasInspector: true,
  hasLibrary: true,
  hasMoveable: false,
  hasPage: true,
  ...
};
```

It can be done with spreading a default strategy first of course and only overwriting where relevant.

I think this is a cleaner solution than sprinkling a lot of `const inspectorVisible = editorMode !== GRID;` all over the code. In stead we would just have `const inspectorVisible = editorMode.hasInspector;`.

Reply:
@barklund, one thing to clarify: the main nuance in "replay" mode right now is that we have to use a somewhat special markup and hence we also need separate set of templates - similar to `Display` vs `Edit`. I guess we could simply have `elementTemplate` property/enum that would indicate which template we use in which mode?

Beyond that, I'm wondering if we mode with page grid view is really a special mode or an overlay above the editor itself (e.g. kind of like a dialog). Do you have reasoning one way or another for this?

Reply:
@dvoytenko, I don't have strong feelings on way or the other about either. They could probably be either two different small local states or both be "big" editor-wide view modes. I just wanted to avoid have a lot of constants and a lot of if's showing and hiding things throughout the application.

Thinking more closely about it, and can't really come up with any other global state like this. Maybe if we had something akin to "switch template", that would allow the user to "go back" to template selection and apply a different one, but that's only in the far future if ever.

So any implementation here is fine with me. When @swissspidy gets to implement grid view, he can figure out the best way to implement the state for that.

Reply:
Dialog seems reasonable. Just saw that the spreadsheet refers to it as a modal too FWIW.